duo_client/paginator.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import absolute_import, print_function
+from time import sleep
+import six
+from .exceptions import MaxRequestAttemptsReached
+
+
+def get_backoff_seconds_exp(wait_const, min_wait_seconds, max_attempts):
+    """Calculate a counter and wait time in seconds for each retried request.
+
+    :param wait_const: The constant to apply to the exponential calculation
+    :param min_wait_seconds: Minimum wait time to yield
+    :max_attemps: Maximum number of wait times to yield
+    :returns: A generator that yields a counter and a wait time in seconds
+    :rtype: generator
+    """
+
+    for attempt in range(max_attempts):
+        yield (attempt, max(wait_const**attempt, min_wait_seconds))
+
+
+class Paginator(object):
```

Comment:
I wonder if we should rename this to BackOffPaginator or something like that.
Just because we already do have Pagination without using this class?

Reply:
I think that's a good idea Xander, it'll make a clearer distinction between those modules. 👍